cdap-tms/src/main/java/co/cask/cdap/messaging/service/CoreMessagingService.java

```diff
@@ -247,6 +256,29 @@ protected void startUp() throws Exception {
     LOG.info("Core Messaging Service started");
   }
 
+  /**
+   * Creates the given topic if it is not yet created. Adds a topic to the asyncCreationTopics if creation fails.
+   */
+  private void createSystemTopic(String topicName,  Queue<TopicId> asyncCreationTopics) {
+    TopicId topicId;
+    try {
+      topicId = NamespaceId.SYSTEM.topic(topicName);
+    } catch (IllegalArgumentException e) {
+      // Ignore invalid topic
+      LOG.warn("Ignore creation of invalid topic '{}'.", topicName);
```

Comment:
this was existing code, but the message should also be improved. Should be error, as some functionality will be broken, and it should mention why it's an invalid id. The exception message mentions which characters are allowed. `'{}'. {}", topicName, e.getMessage()`.